packages/api/src/hooks/useJurisdictionStatus.ts

```diff
@@ -0,0 +1,108 @@
+
+/**
+ * Hook that returns the verification status for a landing company/jurisdiction
+ *
+ * @param jurisdiction - the jurisdiction/landing company, i.e. 'svg', 'bvi', 'labuan', 'vanuatu'
+ * @param mt5_account_status - the status of the MT5 account, usually retrieved from MT5 accounts list item
+ * @returns One of the following statuses: failed, pending, verified, and not_applicable. not_applicable is only applied for SVG jurisdiction.
+ */
+const useJurisdictionStatus = (
+    jurisdiction: TAccount['landing_company_short'],
+    mt5_account_status: TAccount['status']
+) => {
+    const { data: authenticationStatus, isSuccess: isSuccessAuthenticationStatus } = useAuthentication();
+    const { data: accountStatus, isSuccess: isSuccessAccountStatus } = useAccountStatus();
+    const { data: poiStatus } = usePOI();
+    const { data: poaStatus, isSuccess: isSuccessPOA } = usePOA();
+
+    const isSuccess = useMemo(() => {
+        return isSuccessAuthenticationStatus && isSuccessAccountStatus && isSuccessPOA && poiStatus?.next?.service;
+    }, [isSuccessAuthenticationStatus, isSuccessAccountStatus, poiStatus?.next?.service, isSuccessPOA]);
+
+    const verification_status = useMemo(() => {
+        const isServiceStatus = (...statuses: TServiceStatus[]) => {
+            const next_service = poiStatus?.next?.service as keyof TServices;
+            const service = poiStatus?.services?.[next_service];
+            if (service?.status) {
+                return statuses.includes(service.status);
+            }
+        };
+
+        const is_poa_failed = poaStatus?.is_rejected || poaStatus?.is_suspected || poaStatus?.is_expired;
+
+        let status: TJurisdictionStatuses;
+        switch (jurisdiction) {
+            case 'bvi':
+                if (
+                    isServiceStatus('expired', 'none', 'rejected', 'suspected') ||
+                    mt5_account_status === 'proof_failed'
+                ) {
+                    status = 'failed';
+                } else if (isServiceStatus('pending') || mt5_account_status === 'verification_pending') {
+                    status = 'pending';
+                } else {
+                    status = 'verified';
+                }
+                break;
+            case 'labuan':
+                // for labuan we check POA
+                // for labuan we just need to check for is_authenticated_with_idv_photoid status for failed case
+                if (
+                    is_poa_failed ||
+                    isServiceStatus('expired', 'none', 'rejected', 'suspected') ||
+                    authenticationStatus?.is_authenticated_with_idv_photoid ||
+                    mt5_account_status === 'proof_failed'
+                ) {
+                    status = 'failed';
+                } else if (isServiceStatus('pending') || mt5_account_status === 'verification_pending') {
+                    status = 'pending';
```

Comment:
whats the difference between original "status" and new "status"? it look really confusing when one status with specific set of values is mapped to another "status" with different set of values... I can imagine this being pretty tough to debug

Reply:
It's a status for the verification badge only which only consists of `failed | pending | verified | not_supported`, but I can see this could be confusing as to the intent of this status and could mess up the types for the consumer's part 😅 
Perhaps I should refactor it to return the status values itself, something like:
```
return {
   is_failed,
   is_pending,
   is_verified,
   is_not_supported
}
```

Reply:
Yes, i think it could be better, but I do not want to jump on quick conclusions, as complex topics like this requires really good thinking how to separate concerns to make it readable and maintanable.

Alternatively - instead of "status" use name which reflects the actual meaning (if finding something is hard, then it does suggest to me that it worth rethinking the approach to the topic)